### ISSUE TITLE
Unable to set vmin vmax when plotting vector data

### ISSUE THREAD

**Issue by wangjiawen2013** [scverse/spatialdata-plot#324]
Hi,
I want set the vmin and vmax of the color, but failed. This is the first method:
```
norm = matplotlib.colors.Normalize(vmin=-1.0, vmax=0.4)
sdata.pl.render_shapes("cell_circles", norm=norm, color="NOTCH2", method="matplotlib").pl.show(coordinate_systems= "physical", figsize=(10,4))
```
The resulting plot is here:
![image](https://github.com/user-attachments/assets/ebea5ffd-45bd-4f9d-9b82-4dbb91388dd1)

This is the second method:
`sdata.pl.render_shapes("cell_circles", vmin=-1, vmax=1, color="NOTCH2", method="matplotlib").pl.show(coordinate_systems= "physical", figsize=(10,4))`
The resulting plot is here:
![image](https://github.com/user-attachments/assets/b4668608-eb7e-40b7-9ba9-38310e22324d)

`vmin` and `vmax` didn't work in both cases.

**Comment by LucaMarconato** [comment]
Hi, this is related to this https://github.com/scverse/spatialdata-plot/issues/303, please try using `clip=True`.

**Comment by wangjiawen2013** [comment]
@LucaMarconato 
`clip` still doesn't work, Could you show me an example on how to set colorbar vmin and vmax ?

**Comment by LucaMarconato** [comment]
Thanks for the observation. I have changed the title to emphasize that the issue is also open for vector data (the linked issue is indeed for raster data).
Regarding the missing colorbar, I already reported the issue here: https://github.com/scverse/spatialdata-plot/issues/304, so we can follow up there. The developer assigned to the issue is temporarily not available for the next 14 days because of a pressing deadline.

**Comment by olivermccallion** [comment]
Running into the same problem with spatialdata-plot version 0.2.6.dev3+gffd4a1f where:

```
region.pl.render_shapes(
    "cell_boundaries", color = "CD3E", vmax = 11, clip=True).pl.show(title="CD3", coordinate_systems="global", ax=axes[1])
```
```
region.pl.render_shapes(
    "cell_boundaries", color = "CD3E", vmax = 11).pl.show(title="CD3", coordinate_systems="global", ax=axes[1])
```
and
```
norm = matplotlib.colors.Normalize(vmin=0.0, vmax=2000)
region.pl.render_shapes(
    "cell_boundaries", color = "CD3E", norm=norm, clip=True).pl.show(title="CD3", coordinate_systems="global", ax=axes[1])
```
do not alter the colorbar limits / shape shading. Is there any update on the best way to fix this?

**Comment by wangjiawen2013** [comment]
@olivermccallion 
Here is a workaround, spatialdata doesn't support vmin and vmax generally:
https://github.com/scverse/spatialdata-plot/issues/304

**Comment by wangjiawen2013** [comment]
Unfortunately not. https://github.com/scverse/spatialdata-plot/issues/304#issuecomment-2227415168

**Comment by wangjiawen2013** [comment]
@timtreis @olivermccallion 
I have tested the latest spatialdata-plot (v0.2.6)
`vmin` and `mvax` works only in `pl.render_images`, but still don't work in `pl.render_shapes` .

**Comment by timtreis** [comment]
I'm surprised it even works in render_images :D Thought I had removed it. Could you try creating a matplotlib Normalise object with your desired vmin/vmax values and pass that to `norm`? That should work

**Comment by LucaMarconato** [comment]
@timtreis I would either remove `vmin` `vmax`, or create a `Normalise` object if `vmin` `vmax` are passed.

**Comment by wangjiawen2013** [comment]
I think you should keep `vmin` and `vmax`. We can use these two parameters to clip/truncate the colorbar to make the figure more saturated. It's different from `norm`. We use `norm` to normalize the colorbar, but don't clip/truncate it. So `vmin/vmax` have different functions with `norm`.

**Comment by wangjiawen2013** [comment]
> I'm surprised it even works in render_images :D Thought I had removed it. Could you try creating a matplotlib Normalise object with your desired vmin/vmax values and pass that to `norm`? That should work

I created a matplotlib Normalise object with the desired vmin/vmax values and pass that to `norm`, but this is not what I need. What I need is to clip/truncate the colorbar, not normalization.

**Comment by timtreis** [comment]
Hey @wangjiawen2013,

if I understand correctly, this is your usecase?

<img width="876" alt="image" src="https://github.com/user-attachments/assets/f94261b2-3497-41fc-861b-5c51f3a09684">


Or did I misunderstand something?

**Comment by timtreis** [comment]
The current `norm` based approach allows you to make use of the other `mpl.colors` Normalisation methods, like f.e. this:

<img width="693" alt="image" src="https://github.com/user-attachments/assets/abc3cc2a-ee59-4aab-9bae-7b647906dd4c">

**Comment by wangjiawen2013** [comment]
@timtreis 
Yes, this is what I mean !  I am not familiar with matplotlib `norm` and don't know the `clip` parameter. I tested my data and found this works in `pl.render_shapes`, while don't work for `pl.render_shapes`, could you extend `norm` to pl.render_shapes` ?

**Comment by timtreis** [comment]
Oh weird, I'd think it should also work for `render_shapes`. I'll investigate and ping you

**Comment by melonora** [comment]
Picking this up today as my first PR getting back to development hehe. Will report back

**Comment by melonora** [comment]
I take it that we should perhaps also add a notebook explaining the matplotlib norm.

**Comment by LucaMarconato** [comment]
Agree. We can add this to the [static plot workshop notebook](https://github.com/PMBio/spatialdata-workshops/blob/main/notebooks/nb2_static_plotting.ipynb) and maybe a few extra notebooks in the `spatialdata` repo.

**Comment by melonora** [comment]
ok fixed it, just testing now, but here already with vmin / vmax / clip and without:
![image](https://github.com/user-attachments/assets/5ff27cb8-0c70-4aa7-8a59-f919f7d53fd6)
![image](https://github.com/user-attachments/assets/52927160-0931-4137-918a-3f30138c185d)

**Comment by melonora** [comment]
@wangjiawen2013 if you would like you can check #368 to see whether it fixes the issue for you. This only fixes it for the matplotlib render. Once the solution is in I will also create a fix for the datashader.

**Comment by wangjiawen2013** [comment]
Hi,
I am trying to test it. Is this fixes released in current spatialdata version ?

**Comment by melonora** [comment]
Hi! This is not in the current release yet I believe. Thanks for your willingness to test! If you want you can test from main.

**Comment by timtreis** [comment]
Hey @wangjiawen2013, it's not yet released but we'll have one this week that also includes a bunch of datashader fixes :)
